### PR TITLE
#36 Ask User for Name

### PR DESCRIPTION
**Pull request by brucegomes**
* If Name is null when joining a room we display a popup asking the user for a name and joining them to the room

### CODE REVIEW

File: PointerStar/Client/Pages/Room.razor [hunk 1]
```diff
@@ -84,7 +144,27 @@ else if (ViewModel.IsTeamMember)
...
+                Open();
+            }
+            else
+            {
+                await joinRoom();
+            }
+        }
+    }
+
+    private async Task joinRoom()
```
Comment (Keboo) [inline PointerStar/Client/Pages/Room.razor:158]:
```suggestion
    private async Task JoinRoom()
```

File: PointerStar/Client/Pages/Room.razor [hunk 2]
```diff
@@ -173,7 +233,27 @@
         await base.OnInitializedAsync();
         if (RoomId is { } roomId)
         {
-            User user = new(Guid.NewGuid(), Name ?? $"User {new Random().Next()}");
+            if(string.IsNullOrEmpty(Name))
```
Comment (Keboo) [inline PointerStar/Client/Pages/Room.razor:147]: Should this be moved into the ViewModel? There is an `OnInitializedAsync` method that can be overriden there.

File: Tests/PointerStar.Client.Tests/PointerStar.Client.Tests.csproj [hunk 3]
```diff
@@ -4,6 +4,10 @@
     <TargetFramework>net7.0</TargetFramework>
   </PropertyGroup>
 
+  <ItemGroup>
+    <PackageReference Include="Majorsoft.Blazor.Components.Modal" Version="1.5.0" />
```
Comment (Keboo) [inline Tests/PointerStar.Client.Tests/PointerStar.Client.Tests.csproj:8]: Do we need this in the shared project? I would only expect it to be needed in the Client project

File: PointerStar/Shared/PointerStar.Shared.csproj [hunk 4]
```diff
@@ -5,6 +5,7 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Include="Majorsoft.Blazor.Components.Modal" Version="1.5.0" />
```
Comment (Keboo) [inline PointerStar/Shared/PointerStar.Shared.csproj:8]: Do we need this in the Server project? I would only expect it to be needed in the Client project

File: PointerStar/Client/Pages/Room.razor [hunk 5]
```diff
@@ -10,157 +10,128 @@
...
-                    </div>
-                    <div class="form-check form-switch" style="margin-left:10px">
-                        <input class="form-check-input" type="checkbox" @bind="ViewModel.PreviewVotes">
-                        <label class="form-check-label" for="flexSwitchCheckDefault">Preview votes</label>
-                    </div>
-                </div>
-                <div>
-                    <button class="btn btn-warning" @onclick="ViewModel.ResetVotesAsync">Reset Votes</button>
-                </div>
+@if (ViewModel.IsFacilitator)
```
Comment (Keboo) [inline PointerStar/Client/Pages/Room.razor:13]: It looks like you might have changed the layout after my last PR merged.

File: PointerStar/Client/Pages/Room.razor [hunk 6]
```diff
@@ -173,7 +144,27 @@
...
+    }
+
+    private async Task JoinRoom()
+    {
+        if(IsModalOpen)
+        {
+            Close();
+        }
+
+        User user = new(Guid.NewGuid(), Name ?? $"User {new Random().Next()}");
```
Comment (Keboo) [inline PointerStar/Client/Pages/Room.razor:165]: Should the user's name be used here?

File: Tests/PointerStar.Server.Tests/PointerStar.Server.Tests.csproj [hunk 7]
```diff
@@ -5,6 +5,7 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Include="Majorsoft.Blazor.Components.Modal" Version="1.5.0" />
```
Comment (Keboo) [inline Tests/PointerStar.Server.Tests/PointerStar.Server.Tests.csproj:8]: Do we need this in the tests project? I would only expect it to be needed in the Client project

File: Tests/PointerStar.Shared.Tests/PointerStar.Shared.Tests.csproj [hunk 8]
```diff
@@ -4,6 +4,10 @@
     <TargetFramework>net7.0</TargetFramework>
   </PropertyGroup>
 
+  <ItemGroup>
+    <PackageReference Include="Majorsoft.Blazor.Components.Modal" Version="1.5.0" />
```
Comment (Keboo) [inline Tests/PointerStar.Shared.Tests/PointerStar.Shared.Tests.csproj:8]: Do we need this in the shared project? I would only expect it to be needed in the Client project

File: PointerStar/Server/PointerStar.Server.csproj [hunk 9]
```diff
@@ -6,6 +6,7 @@
 
   <ItemGroup>
     <PackageReference Include="Hashids.net" Version="1.6.1" />
+    <PackageReference Include="Majorsoft.Blazor.Components.Modal" Version="1.5.0" />
```
Comment (Keboo) [inline PointerStar/Server/PointerStar.Server.csproj:9]: Do we need this in the Server project? I would only expect it to be needed in the Client project

File: PointerStar/Client/PointerStar.Client.csproj [hunk 10]
```diff
@@ -7,8 +7,13 @@
 
   <ItemGroup>
     <PackageReference Include="CommunityToolkit.Mvvm" Version="8.0.0" />
+    <PackageReference Include="Majorsoft.Blazor.Components.Modal" Version="1.5.0" />
+    <PackageReference Include="Microsoft.AspNet.Razor" Version="3.2.9" />
     <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly" Version="7.0.1" />
     <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly.DevServer" Version="7.0.1" PrivateAssets="all" />
+    <PackageReference Include="Microsoft.AspNetCore.Razor.Runtime" Version="2.2.0" />
+    <PackageReference Include="Microsoft.CodeAnalysis.Razor" Version="6.0.12" />
+    <PackageReference Include="RazorEngine" Version="3.10.0" />
```
Comment (Keboo) [inline PointerStar/Client/PointerStar.Client.csproj:16]: Do we need the RazorEngine package?

File: PointerStar/Client/PointerStar.Client.csproj [hunk 11]
```diff
@@ -7,8 +7,13 @@
 
   <ItemGroup>
     <PackageReference Include="CommunityToolkit.Mvvm" Version="8.0.0" />
+    <PackageReference Include="Majorsoft.Blazor.Components.Modal" Version="1.5.0" />
+    <PackageReference Include="Microsoft.AspNet.Razor" Version="3.2.9" />
     <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly" Version="7.0.1" />
     <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly.DevServer" Version="7.0.1" PrivateAssets="all" />
+    <PackageReference Include="Microsoft.AspNetCore.Razor.Runtime" Version="2.2.0" />
```
Comment (Keboo) [inline PointerStar/Client/PointerStar.Client.csproj:14]: Do we need the `Microsoft.AspNetCore.Razor.Runtime` package?

File: PointerStar/Client/Pages/Room.razor (hunk 6 above)
Comment (brucegomes) [reply PointerStar/Client/Pages/Room.razor:165]: @Keboo  it is being used. the `@bind="Name"` in line 172 mean anythign typed in the input box goes into the Name property. continue this on PR !51 If you need to

**Comment by brucegomes** [comment]
closing because this PR got updated with #51

**Pull request closed without merging**